File: lock-packagelock/handler.js
'use strict'

module.exports = async (event, context) => {
    const result = {
        'dependencies': []
    }

    let lockfile = JSON.parse(event.body);

    Object.keys(lockfile.dependencies).forEach( (elem,index) => {
        result['dependencies'].push( [ elem, lockfile.dependencies[elem].version ] )
    });

    return context
        .status(200)
        .succeed(result)
}

File: lock-gemfile/handler.rb
require 'bundler'
require 'json'

class Handler
  def run(body)
    result = {dependencies: []}

    context = Bundler::LockfileParser.new( body )

    context.specs.each do |s|
      result[:dependencies] << [s.name, s.version.to_s]
    end

    return result.to_json
  end
end
